Package SDK Tests as global tools

dsplaisted opened the pull request:
This creates global tools packages for the SDK tests.  This should enable:

- Running tests against an installed version of the .NET Core SDK.  This should help protect against bugs such as #2128 (where we were trying to write to a location that is write-protected if the SDK is installed normally, but not in the setup we have to run tests in the repo).
- Automating our perf tests to run against the dotnet/cli repo and/or ProdCon builds

dsplaisted committed:
Support specifying the .NET Core SDK version to use via test command line

dsplaisted committed:
Fix duplicate file warnings for packing tests as tool

dsplaisted committed:
Add *.binlog to .gitignore

dsplaisted committed:
Reuse resource file for using strings in tests instead of referencing Microsoft.NET.Build.Tasks project

dsplaisted committed:
Enable packing test projects

dsplaisted committed:
Move TelemetryStdOutLogger to separate project

This enables turning off CopyLocalLockFileAssemblies in Microsoft.NET.Build.Tests, which causes issues with packing the project as a tool

dsplaisted committed:
Retarget tests to netcoreapp2.1

dsplaisted committed:
Include test assets in test tool package, and update TestContext to support tests as tools

dsplaisted committed:
Use short names for test tools

dsplaisted committed:
Add -showSdkInfo command line parameter to tests

dsplaisted committed:
Move test packages to subfolder

dsplaisted commented on src/Directory.Build.targets:
```diff
@@ -2,12 +2,30 @@
+    <ToolCommandName>$(PackageId)</ToolCommandName>
+    
+    <!-- Put packages for tests in subfolder so we don't try to sign them -->
+    <PackageOutputPath>$(PackageOutputPath)tests\</PackageOutputPath>
```
We will need to modify the build definition to also push the NuGet packages in this folder.  (If we switch to YAML build definitions changes like this will be a bit easier)

nguerrera commented on src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/Microsoft.NET.Build.Tasks.UnitTests.csproj:
```diff
@@ -38,4 +38,8 @@
 
+  <!-- Workaround for fixed bug causes duplicate file warnings on pack, so disable it
+       https://github.com/dotnet/roslyn-tools/issues/249 -->
+  <Target Name="WorkaroundIncorrectSatelliteDeployment" />
```
You're going to get the repo toolset fix in before this, right?

dsplaisted replied:
@nguerrera I've merged https://github.com/dotnet/roslyn-tools/pull/253, and updated this PR to use the new repo toolset and remove the workaround.

dsplaisted committed:
Update to new RepoToolset, import it from Directory.Build files instead of each project file

nguerrera commented on src/Tests/Microsoft.NET.TestFramework/TestContext.cs:
```diff
@@ -71,13 +71,37 @@ public static void Initialize(TestCommandLine commandLine)
+            if (runAsTool)
+            {
+                testContext.TestExecutionDirectory = Path.Combine(Path.GetTempPath(), "dotnetSdkTests");
+                
```
nit: extra newline

dsplaisted committed:
Update expected roll-forward versions

nguerrera commented on src/Tests/Directory.Build.targets:
```diff
@@ -0,0 +1,22 @@
+
+  <ItemGroup Condition="'$(IsTestProject)' == 'true' AND '$(OutputType)' == 'Exe'">
+
+    <!-- Note: These items are showing up in solution explorer even though Visible is set to false -->
```
Is there a project system bug tracking this? cc @davkean

davkean replied:
I don't know what the bug is - the items show up? Or just the folder?

dsplaisted replied:
@davkean Turns out it's just the folders that show up:

![image](https://user-images.githubusercontent.com/145043/39949273-b6f68faa-552e-11e8-8fdb-a0c8f47df456.png)

davkean replied:
Yeah we have a bug on this.

dsplaisted committed:
Don't pack unit tests as tools

dsplaisted committed:
Update repo toolset

This should fix issues with sourcelink on Linux

dsplaisted committed:
Remove Directory.Build.props and .targets from repo root.

This works around #2240, which is causing GivenThatWeWantToGenerateADepsFileForATool tests to fail
